coreax/metrics.py

```diff
@@ -119,13 +119,14 @@ def compute(
         del kwargs
         _block_size = coreax.util.tree_leaves_repeat(block_size, 2)
         bs_xx, bs_xy, _, bs_yy = tuple(product(_block_size, repeat=len(_block_size)))
+        u_x, u_y = coreax.util.tree_leaves_repeat(unroll, 2)
```

Comment:
For purely stylistic reasons, I think it would be better if we exactly matched the block_size case. 
```python
_unroll = coreax.util.tree_leaves_repeat(unroll, 2)
u_xx, u_xy, _, u_yy = tuple(product(_block_size, repeat=len(_unroll)))
```
Not that it really matters, but from the perspective of compilation time, we could slightly reduce compilation time here by using a scan, which also includes the `compute_mean` lines. Something like:
```python
_block_size = coreax.util.tree_leaves_repeat(block_size, 2)
_unroll = coreax.util.tree_leaves_repeat(unroll, 2)
_data = reference_data, comparison_data
indices = [(0,0),(0,1),(1,1)] # i_xx, i_xy, i_yy

def _mean(_, val):
    i_x, i_y = val
    (x, y), bs, un = [(z[i_x], z[i_y]) for z in (_data, _block_size, _unroll)]
    return _, self.kernel.compute_mean(x, y, block_size=bs, unroll=un)

_, [k_xx_mean, k_xy_mean, k_yy_mean] = jax.lax.scan(_mean, None, indices)
squared_mmd_threshold_applied = coreax.util.apply_negative_precision_threshold(
    k_xx_mean + k_yy_mean - 2 * k_xy_mean,
    self.precision_threshold,
)
```
If you want to time this, I would be interested to see if there is an appreciable difference. If not, then I'm happy to leave this as a future ticket. Note: assume compilation time is roughly the difference in runtime between the first and second call of a JIT'ed function.

Reply:
I've made that stylistic change, probably worth making a new ticket for the compilation time Q